File: uhrwerk-common/src/main/java/io/qimia/uhrwerk/common/model/TargetModelBuilder.java
package io.qimia.uhrwerk.common.model;

import org.jetbrains.annotations.NotNull;

public class TargetModelBuilder extends StateModelBuilder {

  Long id;
  Long tableId;
  Long connectionId;
  ConnectionModel connection;
  TableModel table;

  String format;
  boolean active = true;

  public TargetModelBuilder id(Long id) {
    this.id = id;
    return this;
  }

  public TargetModelBuilder tableId(@NotNull Long tableId) {
    this.tableId = tableId;
    return this;
  }

  public TargetModelBuilder connectionId(@NotNull Long connectionId) {
    this.connectionId = connectionId;
    return this;
  }

  public TargetModelBuilder connection(ConnectionModel connection) {
    this.connection = connection;
    return this;
  }

  public TargetModelBuilder table(TableModel table) {
    this.table = table;
    return this;
  }

  public TargetModelBuilder format(@NotNull String format) {
    this.format = format;
    return this;
  }

  public TargetModel build() {
    return new TargetModel(this);
  }

  @Override
  StateModelBuilder getThis() {
    return this;
  }
}

File: uhrwerk-metastore/src/main/kotlin/io/qimia/uhrwerk/common/metastore/builders/TargetModelBuilder.kt
package io.qimia.uhrwerk.common.metastore.builders

import io.qimia.uhrwerk.common.metastore.model.ConnectionModel
import io.qimia.uhrwerk.common.metastore.model.TableModel
import io.qimia.uhrwerk.common.model.*

class TargetModelBuilder : StateModelBuilder<TargetModelBuilder>() {
    var id: Long? = null
    var tableId: Long? = null
    var connectionId: Long? = null
    var connection: ConnectionModel? = null
    var table: TableModel? = null
    var format: String? = null
    var tableName: String? = null
    fun id(id: Long?): TargetModelBuilder {
        this.id = id
        return this
    }

    fun tableId(tableId: Long): TargetModelBuilder {
        this.tableId = tableId
        return this
    }

    fun connectionId(connectionId: Long): TargetModelBuilder {
        this.connectionId = connectionId
        return this
    }

    fun connection(connection: ConnectionModel?): TargetModelBuilder {
        this.connection = connection
        return this
    }

    fun table(table: TableModel?): TargetModelBuilder {
        this.table = table
        return this
    }

    fun format(format: String): TargetModelBuilder {
        this.format = format
        return this
    }

    fun tableName(tableName: String?): TargetModelBuilder {
        this.tableName = tableName
        return this
    }

    fun build(): TargetModel {
        return TargetModel(
            id,
            format,
            tableId,
            connectionId,
            tableName,
            connection,
            table
        )
    }

    override fun getThis() = this

}